src/os/memstore/MemStore.cc

```diff
@@ -1384,6 +1384,13 @@ int MemStore::_destroy_collection(const coll_t& cid)
     cp->second->exists = false;
   }
   used_bytes -= cp->second->used_bytes();
+
+  // remove data file if it is existed
+  string fn = path + "/" + stringify(cp->first);
+  int r = ::unlink(fn.c_str());
+  if (r < 0 && r != -ENOENT)
+    return r;
+
```

Comment:
I don't think this should be guarded by new option.  The case is:

1) create pool with pgs,  memstore_persistent_collections = true now
2) then stop the osd, and the collection file will be saved
3) start osd then with memstore_persistent_collections = false
4) delete the pool and destroy pg

if guarded by option, the stale *_head will stay in the data_dir